### PR TITLE
Fix onion packet lock time & forward amounts and verify HTLCs

### LINKED ISSUE #1683: Flash: Verify HTLCs with the encrypted packet parameters
Currently they are just assumed to be correct.

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Requires https://github.com/bpfkorea/agora/pull/1682 which still seems to be timing out because of some reasons.

Please ignore the first commit.

Fixes #1683

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
> Error relocating /usr/local/bin/agora: _D6object6Object6toHashMFNbNeZm: symbol not found

What the..

**Comment by AndrejMitrovic** [comment]
I'm rebasing the `flash-layer` branch against `v0.x.x`. I'll push that soon.

File: source/agora/flash/OnionPacket.d [hunk 1]
```diff
@@ -200,12 +200,11 @@ public OnionPacket createOnionPacket (in Hash payment_hash,
 
         packet.encrypted_payloads[last_index] = encrypted_payload;
         last_index--;
-        if (!forward_amount.sub(hop.fee))
+        if (!forward_amount.add(hop.fee))
```
Comment (omerfirmak) [inline source/agora/flash/OnionPacket.d:203]: I think we have `mustAdd/mustSub` for use cases like this.

**Review by omerfirmak — APPROVED** [review]

File: source/agora/flash/OnionPacket.d (hunk 1 above)
Comment (Geod24) [reply source/agora/flash/OnionPacket.d:203]: https://github.com/bpfkorea/agora/blob/df511f4cbfaf6cca04b92fbe1be5175eac85911f/source/agora/common/Amount.d#L301

**Pull request merged**